x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -189,6 +189,9 @@ public Realm realm(String name) {
             if (factory == null) {
                 throw new IllegalArgumentException("unknown realm type [" + identifier.getType() + "] for realm [" + identifier + "]");
             }
+            if (realms.stream().anyMatch(r -> r.name().equals(identifier.getName()))) {
+                throw new IllegalArgumentException("multiple realms configured with the same name [" + identifier.getName() + "] ");
```

Comment:
I think this needs to be a little more verbose - perhaps output the full-setting-key for the duplicate names?
```
found [xpack.security.authc.realms.ldap.my_realm] and [xpack.security.authc.realms.pki.my_realm]
```